nixos/modules/services/web-apps/privatebin.nix

```
@@ -0,0 +1,278 @@
+        for details on configuration directives.
+      '';
+    };
+
+    phpPackage = mkOption {
+      type = types.package;
+      relatedPackages = [ "php80" "php81" "php82" ];
+      default = pkgs.php81;
+      defaultText = "pkgs.php81";
+      description = lib.mdDoc ''
+        PHP package to use for this PrivateBin instance.
+      '';
+    };
+
+    phpOptions = mkOption {
+      type = types.attrsOf types.str;
+      default = { };
+      description = lib.mdDoc ''
+        Options for PHP's php.ini file for this dokuwiki site.
+      '';
+      example = literalExpression ''
+        {
+          "opcache.interned_strings_buffer" = "8";
+          "opcache.max_accelerated_files" = "10000";
+          "opcache.memory_consumption" = "128";
+          "opcache.revalidate_freq" = "15";
+          "opcache.fast_shutdown" = "1";
+        }
+      '';
+    };
+
+    nginx = mkOption {
+      type = types.submodule (import ../web-servers/nginx/vhost-options.nix { inherit config lib; });
+      default = {};
+      example = literalExpression ''
+        {
+          forceSSL = true;
+          enableACME = true;
+        }
+      '';
+      description = lib.mdDoc ''
+        Optional settings to pass to the nginx virtualHost.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    users.users.${user} = {
+      inherit group;
+
+      isSystemUser = true;
+      createHome = false;
+    };
+    users.users.${config.services.nginx.user} = mkIf config.services.nginx.enable { extraGroups = [ group ]; };
+    users.groups.${group} = { };
+
+    services.phpfpm.pools.${user} = {
+      inherit user group;
+      settings = {
+        "listen.mode" = "0660";
+        "listen.owner" = user;
+        "listen.group" = group;
+        "catch_workers_output" = true;
+        "clear_env" = false;
+      } // cfg.poolConfig;
+      phpEnv.CONFIG_PATH = "${configFile}";
+    };
+
+    services.postgresql = mkIf (autoDb == "pgsql") {
+      enable = true;
+      ensureUsers = [{
+        name = user;
+        ensurePermissions = {
+          "DATABASE \"${user}\"" = "ALL PRIVILEGES";
+        };
+      }];
+      ensureDatabases = [
+        user
+      ];
+    };
+
+    services.mysql = mkIf (autoDb == "mysql") {
+      enable = true;
+      package = mkDefault pkgs.mariadb;
+      ensureUsers = [{
+        name = user;
+        ensurePermissions = {
+          "${user}.*" = "ALL PRIVILEGES";
+        };
+      }];
+      ensureDatabases = [
+        user
+      ];
+    };
+    systemd.services."phpfpm-${user}".serviceConfig = {
+      ProtectSystem = "full";
+      PrivateTmp = true;
+      EnvironmentFile = cfg.environmentFiles;
+    };
+    systemd.tmpfiles.rules = flatten [
+      "d /var/lib/privatebin/data 0750 ${user} ${group} - -"
+    ];
+
+    services.nginx = {
+      enable = mkDefault true;
+      virtualHosts.privatebin = mkMerge [
+        cfg.nginx
+        {
+          root = mkForce "${pkgs.privatebin}/share/privatebin";
```

```suggestion
          root = mkForce pkgs.privatebin;
```
we could also just place it here like other php web things